onnxscript/evaluator.py

```diff
@@ -116,10 +128,25 @@ class Evaluator(abc.ABC):
 
     def eval(
         self,
-        schema: onnx.defs.OpSchema,
-        inputs: Sequence[ExtendedModeValue],
-        attributes: Mapping[str, Any],
+        op: values.Op,
+        args: Sequence[ExtendedModeValue],
+        kwargs: Mapping[str, Any],
     ):
+        """Evaluates an ONNX op.
+
+        Args:
+            op: The op to evaluate.
+            args: The positional arguments to the op.
+            kwargs: The keyword arguments to the op.
+        """
+        param_schemas = op.param_schemas()
+        # Split happens in the evaluator instead of the Op __call__ method
```

Comment:
It just struck me that perhaps we are likely getting this for free already, because of the current generation of python functions from the operator schema (at least for ONNX ops)? I mean the generated code such as https://github.com/microsoft/onnx-script/blob/9624c4e177351fe49243c1106532dc79a0e66301/onnxscript/onnx_opset/_impl/opset18.py#L143

Reply:
I think eager calls follow a different route?

Reply:
Aka we start from an Op object: `op(*self._prepare_inputs(schema, X))`?

Reply:
Ohh I may be seeing what you are saying

Reply:
In that case we will only care about functions and not built in ops?

Reply:
Done. Removed the call in `eval()` and resumed its signature. Added eager tests.